spyder/plugins/ipythonconsole/widgets/debugging.py

```diff
@@ -92,6 +135,9 @@ def callback(line):
                 if "do_" + cmd not in dir(pdb.Pdb):
                     self._control.history.append(line)
 
+            # must match ConsoleWidget.do_execute
+            self._executing = True
```

Comment:
This is already an attribute present in qtconsole:

https://github.com/jupyter/qtconsole/blob/f9ee7285f5ea6bd4a4aa411dbaae279dac43ac5d/qtconsole/console_widget.py#L266

so I think you don't need to declare it here again.

Reply:
I'm not declaring it, I'm setting it to True to have correct printing behaviour. Otherwise the messages printed during execution might not be displayed.

Reply:
Which breaks some tests sometimes.

Reply:
Ok, thanks for the clarification.